### PR TITLE
devSandbox

### PR DESCRIPTION
**Pull request by marcobarilari**
see Issue #24  for discussion

### CODE REVIEW

**Comment by marcobarilari** [comment]
I think I am done with prettifying this code, LMK what you think. Do you you also have any suggestion for the name?

File: README.md [hunk 1]
```diff
@@ -140,6 +140,24 @@ Define the parameters of the fixation cross in `cfg` and `expParameters` and thi
...
 
+## Annexes
+
+### Experiment template [ WIP ]
+
+### `devSandobox.m` stand-alone
+
+This script is a stand-alone function that can be useful as a sandbox to develop the PTB audio/visual stimulation of your experiment. No input/output required.
+
+Here, a tutorial from https://peterscarfe.com/contrastgratingdemo.html is provided for illustrative purpose (notice that some vars' name are updated to our code style). For your use, you will delete that part.
```
Comment (Remi-Gau) [inline README.md:151]:
```suggestion
Here, a tutorial from https://peterscarfe.com/contrastgratingdemo.html is provided for illustrative purpose (notice that some variable names are updated to our code style). For your use, you will delete that part.
```

File: README.md [hunk 2]
```diff
@@ -140,6 +140,24 @@ Define the parameters of the fixation cross in `cfg` and `expParameters` and thi
...
+
+It is composed of two parts:
+ - a fixed structure that will initialize and close PTB in 'debug mode'
+    (`PsychDebugWindowConfiguration`, `SkipSyncTests`)
+ - the actual sandbox where to set your dynamic vars (the stimualtion
+   parameters) and the 'playground' where to develop the stimulation code
+
+ When you are happy with it, ideally you will move the vars in `setParameters.m` and the stimulation code in a separate function in `my-experiment-folder/subfun`. The code style and vars name is the same used in `cpp-lln-lab/CPP_PTB` github repo, therefore it should be easy to move everything in your experiment scripts (see the template that is annexed in `cpp-lln-lab/CPP_PTB`).
```
Comment (Remi-Gau) [inline README.md:159]:
```suggestion
 When you are happy with it, ideally you will move the vars in `setParameters.m` and the stimulation code in a separate function in `my-experiment-folder/subfun`. The code style and variable names are the same used in `cpp-lln-lab/CPP_PTB` github repo, therefore it should be easy to move everything in your experiment scripts (see the template that is annexed in `cpp-lln-lab/CPP_PTB`).
```

File: README.md [hunk 3]
```diff
@@ -140,6 +140,24 @@ Define the parameters of the fixation cross in `cfg` and `expParameters` and thi
...
+### `devSandobox.m` stand-alone
+
+This script is a stand-alone function that can be useful as a sandbox to develop the PTB audio/visual stimulation of your experiment. No input/output required.
+
+Here, a tutorial from https://peterscarfe.com/contrastgratingdemo.html is provided for illustrative purpose (notice that some vars' name are updated to our code style). For your use, you will delete that part.
+
+It is composed of two parts:
+ - a fixed structure that will initialize and close PTB in 'debug mode'
+    (`PsychDebugWindowConfiguration`, `SkipSyncTests`)
+ - the actual sandbox where to set your dynamic vars (the stimualtion
```
Comment (Remi-Gau) [inline README.md:156]:
```suggestion
 - the actual sandbox where to set your dynamic variables (the stimulation
```

File: devSandbox.m [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+%  required.
+%
+% Here, a tutorial from https://peterscarfe.com/contrastgratingdemo.html is
+%  provided for illustrative purpose (notice that some vars' name are updated
+%  to our code style). For your use, you will delete that part.
+%
+% It is composed of two parts:
+%  - a fixed structure that will initialize and close PTB in 'debug mode'
+%    (`PsychDebugWindowConfiguration`, `SkipSyncTests`)
+%  - the actual sandbox where to set your dynamic vars (the stimualtion
```
Comment (Remi-Gau) [inline devSandbox.m:14]:
```suggestion
%  - the actual sandbox where to set your dynamic variables (the stimulation
```

File: devSandbox.m [hunk 5]
```diff
@@ -0,0 +1,238 @@
+function devSandbox
+
+% This script is a stand-alone function that can be useful as a sandbox to
+%  develop the PTB audio/visual stimulation of your experiment. No input/output
+%  required.
+%
+% Here, a tutorial from https://peterscarfe.com/contrastgratingdemo.html is
+%  provided for illustrative purpose (notice that some vars' name are updated
```
Comment (Remi-Gau) [inline devSandbox.m:8]:
```suggestion
%  provided for illustrative purpose (notice that some variable names are updated
```

File: devSandbox.m [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+%
+% It is composed of two parts:
+%  - a fixed structure that will initialize and close PTB in 'debug mode'
+%    (`PsychDebugWindowConfiguration`, `SkipSyncTests`)
+%  - the actual sandbox where to set your dynamic vars (the stimualtion
+%    parameters) and the 'playground' where to develop the stimulation code
+%
+% When you are happy with it, ideally you will move the vars in `setParameters.m`
+%  and the stimulation code in a separate function in `my-experiment-folder/subfun`.
+%  The code style and vars name is the same used in `cpp-lln-lab/CPP_PTB`
```
Comment (Remi-Gau) [inline devSandbox.m:19]:
```suggestion
%  The code style and variable names are the same used in `cpp-lln-lab/CPP_PTB`
```

File: devSandbox.m [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+
+% Get the Center of the Screen
+cfg.center = [cfg.winRect(3), cfg.winRect(4)]/2;
+
+% Set up alpha-blending for smooth (anti-aliased) lines
+Screen('BlendFunction', cfg.win, 'GL_SRC_ALPHA', 'GL_ONE_MINUS_SRC_ALPHA');
+
+end
+
+function devSandbox_cleanUp
```
Comment (Remi-Gau) [inline devSandbox.m:208]: You prefer to have a subfunction here rather than a using the default "cleanUp" so that everything is self contained?

**Comment by Remi-Gau** [comment]
I checked that it runs.

OK with me.

Can be merged.

File: devSandbox.m (hunk 7 above)
Comment (marcobarilari) [reply devSandbox.m:211]: yes so that, being a stand-alone, I can copy-paste it in my experiment folder and do version control there

**Pull request merged**